src/commands.rs

```diff
@@ -1,15 +1,199 @@
+    #[serde(default)]
+    pub uptime: Option<String>,
+    #[serde(default)]
+    pub load_avg: Option<String>,
+    #[serde(default)]
+    pub signal_strength: Option<i32>,
+    #[serde(default)]
+    pub interface: Option<String>,
+    #[serde(default)]
+    pub debug: bool,
+    #[serde(default)]
+    pub location: Option<String>,
+    #[serde(default)]
+    pub team: Option<String>,
+    #[serde(default)]
+    pub timezone: Option<String>,
+    #[serde(default)]
+    pub hostname: Option<String>,
+    #[serde(default)]
+    pub ws_open: bool,
+    #[serde(default)]
+    pub status: Option<String>,
+    #[serde(default)]
+    pub last_screenshot: Option<String>,
+    #[serde(default)]
+    pub in_sync: bool,
+    #[serde(default)]
+    pub software_version: Option<String>,
+    #[serde(default)]
+    pub hardware_version: Option<String>,
+    #[serde(default)]
+    pub config: ScreenConfig,
+}
 pub struct ScreenCommand {
     authentication: Authentication,
 }
-
 impl ScreenCommand {
-    fn new(authentication: Authentication) -> Self {
+    pub fn new(authentication: Authentication) -> Self {
         Self { authentication }
     }
+    pub fn list(&self) -> anyhow::Result<Value, CommandError> {
+        let url = self.authentication.config.url.clone() + "/v4/screens";
+        let response = self.authentication.build_client()?.get(url).send()?;
+        if response.status().as_u16() != 200 {
+            return Err(CommandError::WrongResponseStatus(
+                response.status().as_u16(),
+            ));
+        }
+        serde_json::from_str(&response.text().unwrap_or_else(|_| "{}".to_owned()))
+            .map_err(CommandError::ParseError)
+    }
+    pub fn get(&self, id: &str) -> anyhow::Result<Value, CommandError> {
+        let url = self.authentication.config.url.clone() + "/v4/screens?id=eq." + id;
+        let response = self.authentication.build_client()?.get(url).send()?;
+        if response.status().as_u16() != 200 {
+            return Err(CommandError::WrongResponseStatus(
+                response.status().as_u16(),
+            ));
+        }
+
+        serde_json::from_str(&response.text().unwrap_or_else(|_| "{}".to_owned()))
+            .map_err(CommandError::ParseError)
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use crate::authentication::Config;
+    use crate::Authentication;
+    use httpmock::{Method::GET, MockServer};
+
+    use envtestkit::lock::lock_test;
+    use envtestkit::set_env;
+    use std::ffi::OsString;
+    use std::fs;
+
+    use crate::commands::ScreenCommand;
+    use serde_json::Value;
+    use tempdir::TempDir;
+
+    #[test]
+    fn test_list_screens_should_return_correct_screen_list() {
+        let tmp_dir = TempDir::new("test").unwrap();
+        let _lock = lock_test();
+        let _test = set_env(OsString::from("HOME"), tmp_dir.path().to_str().unwrap());
+        fs::write(tmp_dir.path().join(".screenly").to_str().unwrap(), "token").unwrap();
```

Comment:
These looks like a common lines for the authenticated tests.

Reply:
well... ok but the first line is basically a lock:

 let _lock = lock_test();
 let _test = set_env(OsString::from("HOME"), tmp_dir.path().to_str().unwrap());
 
 they need to exist inside the function scope